consensus/src/consensus/head_requests.rs

```diff
@@ -101,8 +104,23 @@ impl<TNetwork: Network + 'static> HeadRequests<TNetwork> {
         include_body: bool,
     ) -> Result<Result<Block, BlockError>, RequestError> {
         network
-            .request::<RequestBlock>(RequestBlock { hash, include_body }, peer_id)
+            .request::<RequestBlock>(
+                RequestBlock {
+                    hash: hash.clone(),
+                    include_body,
+                },
+                peer_id,
+            )
             .await
+            .and_then(|block_res| {
+                // Modify response if the block received does not correspond to the one requested.
+                Ok(block_res.and_then(|mut block| {
+                    if block.hash_cached() != hash {
+                        return Err(BlockError::ResponseHashMismatch);
+                    }
+                    Ok(block)
+                }))
+            })
```

Comment:
Clippy suggestion here
```suggestion
            .map(|block_res| {
                // Modify response if the block received does not correspond to the one requested.
                block_res.and_then(|mut block| {
                    if block.hash_cached() != hash {
                        return Err(BlockError::ResponseHashMismatch);
                    }
                    Ok(block)
                })
            })
```